### PR TITLE
Feature paginate data

### PR DESCRIPTION
**Pull request by condef5**
### Notes
* I added a react-query package to make easier data fetching.
* I added a fake API next to paginate the candidates(this should be removed in the future when the API returns the data with pagination)

### Pending tasks
- [x] Should we paginate the political parties?

### CODE REVIEW

**Comment by lavaldi** [comment]
Part of #9

File: components/Candidates.js [hunk 1]
```diff
@@ -20,21 +26,58 @@ function CandidateCard({ candidate }) {
...
+            <button
+              onClick={() => fetchMore()}
+              disabled={!canFetchMore || isFetchingMore}
+              className="text-neutral-400"
+            >
+              {isFetchingMore ? (
+                "Cargando más..."
+              ) : (
+                <span>
+                  {canFetchMore ? "Cargar más..." : "Nada más que cargar"}
```
Comment (cristianbgp) [inline components/Candidates.js]: Can we do something to know if it's the last page and not show "Nada más que cargar"

Something like this
```diff
diff --git a/components/Candidates.js b/components/Candidates.js
index e506f43..20952fe 100644
--- a/components/Candidates.js
+++ b/components/Candidates.js
@@ -36,6 +36,7 @@ export default function Candidates() {
   } = useInfiniteQuery("candidates", fetchCandidates, {
     getFetchMore: (lastGroup, _allGroups) => lastGroup.nextPage,
   });
+  const isLastPage = data && data[data.length - 1].isLastPage;
 
   return (
     <div className="mt-16">
@@ -61,21 +62,19 @@ export default function Candidates() {
               </Fragment>
             ))}
           </div>
-          <div>
-            <button
-              onClick={() => fetchMore()}
-              disabled={!canFetchMore || isFetchingMore}
-              className="text-neutral-400"
-            >
-              {isFetchingMore ? (
-                "Cargando más..."
-              ) : (
-                <span>
-                  {canFetchMore ? "Cargar más..." : "Nada más que cargar"}
-                </span>
-              )}
-            </button>
-          </div>
+          {!isLastPage && (
+            <div>
+              <button
+                onClick={() => fetchMore()}
+                disabled={!canFetchMore || isFetchingMore}
+                className="text-neutral-400"
+              >
+                {isFetchingMore
+                  ? "Cargando más..."
+                  : canFetchMore && "Cargar más..."}
+              </button>
+            </div>
+          )}
         </>
       )}
     </div>
diff --git a/pages/api/candidates.js b/pages/api/candidates.js
index 3ace7ab..f6eed7b 100644
--- a/pages/api/candidates.js
+++ b/pages/api/candidates.js
@@ -8,8 +8,11 @@ function paginatedData(data, page = 1) {
     page * CONGRESS_LIMIT,
   );
   const nextPage = candidates.length > 0 ? page + 1 : null;
+  const isLastPage =
+    data[data.length - 1].identificador ===
+    candidates[candidates.length - 1].identificador;
 
-  return { candidates, nextPage };
+  return { candidates, nextPage, isLastPage };
 }
 
 // eslint-disable-next-line import/no-anonymous-default-export
```

File: pages/index.js [hunk 2]
```diff
@@ -27,13 +27,10 @@ export async function getServerSideProps() {
     `${process.env.API_URL}/organizaciones-politicas`,
   );
   const politicalParties = await responsePoliticalParties.json();
-  const responseCandidates = await fetch(`${process.env.API_URL}/congresistas`);
```
Comment (cristianbgp) [inline pages/index.js:30]: We could prefetch data and set it on the react-query cache with the configuration for nextjs https://react-query.tanstack.com/docs/guides/ssr#integrating-with-nextjs

File: pages/api/candidates.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    return;
+  }
+
+  fetch(`${process.env.API_URL}/congresistas`)
+    .then((res) => res.json())
+    .then((data) => {
+      cacheData = data;
+      res.statusCode = 200;
+      res.json(paginatedData(cacheData, page));
+    });
```
Comment (cristianbgp) [inline pages/api/candidates.js]: What do you think about using async/await here?

**Review by cristianbgp — APPROVED** [review]
LGTM

**Pull request merged**